cedar-policy-core/src/evaluator/err.rs

```diff
@@ -300,22 +323,14 @@ pub enum EvaluationErrorKind {
 }
 
 /// helper function for pretty-printing type errors
-/// INVARIANT: `expected` must have at least one value
-fn pretty_type_error(expected: &[Type], actual: &Type) -> String {
-    match expected.len() {
-        // PANIC SAFETY, `expected` is non-empty by invariant
-        #[allow(clippy::unreachable)]
-        0 => unreachable!("should expect at least one type"),
-        // PANIC SAFETY. `len` is 1 in this branch
-        #[allow(clippy::indexing_slicing)]
-        1 => format!("type error: expected {}, got {}", expected[0], actual),
-        _ => {
-            use itertools::Itertools;
-            format!(
-                "type error: expected one of [{}], got {actual}",
-                expected.iter().join(", ")
-            )
-        }
+fn pretty_type_error(first_expected: &Type, rest_expected: &[Type], actual: &Type) -> String {
```

Comment:
why `first_expected` and `rest_expected` here, but `expected_first` and `expected_rest` everywhere else?